friends/src/components/PrivateRoute.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+// It has the same API as <Route />.
+// It renders a < Route /> and passes all the props through to it.
+// It checks if the user is authenticated, if they are, it renders the “component” prop.If not, it redirects the user to / login.
+const PrivateRoute = ({ component: Component, ...rest }) => {
+  return (
+    <Route
```

Comment:
Why are you spreading the rest up here ? This is where the private route is breaking. Remember this is a Higher order component,

`private route = component => loginpage =>`